src/main/java/org/broadinstitute/hellbender/tools/walkers/haplotypecaller/graphs/JunctionTreeKBestHaplotypeFinder.java

```diff
@@ -0,0 +1,198 @@
+package org.broadinstitute.hellbender.tools.walkers.haplotypecaller.graphs;
+
+import com.google.common.annotations.VisibleForTesting;
+import org.broadinstitute.hellbender.tools.walkers.haplotypecaller.readthreading.ExperimentalReadThreadingGraph;
+import org.broadinstitute.hellbender.tools.walkers.haplotypecaller.readthreading.MultiDeBruijnVertex;
+import org.broadinstitute.hellbender.utils.Utils;
+
+import java.util.*;
+import java.util.stream.Collectors;
+
+public class JunctionTreeKBestHaplotypeFinder<V extends BaseVertex, E extends BaseEdge> extends KBestHaplotypeFinder<V, E> {
+    public static final int DEFAULT_OUTGOING_JT_EVIDENCE_THRESHOLD_TO_BELEIVE = 3;
+    public static final int DEFAULT_MINIMUM_WEIGHT_FOR_JT_BRANCH_TO_NOT_BE_PRUNED = 2;
+    public static final int DEFAULT_MAX_ACCEPTABLE_DECISION_EDGES_WITHOUT_JT_GUIDANCE = 7;
+    public static final int DEFAULT_MAX_ACCEPTABLE_REPETITIONS_OF_A_KMER_IN_A_PATH = 10;
+    private int weightThresholdToUse = DEFAULT_OUTGOING_JT_EVIDENCE_THRESHOLD_TO_BELEIVE;
+
+    // List for mapping vertexes that start chains of kmers that do not diverge, used to cut down on repeated graph traversal
+    Map<V, List<E>> contiguousSequences = new HashMap<>();
+
+    // Graph to be operated on, in this case cast as an ExperimentalReadThreadingGraph
+    ExperimentalReadThreadingGraph experimentalReadThreadingGraph;
+
+    public JunctionTreeKBestHaplotypeFinder(final BaseGraph<V, E> graph, final Set<V> sources, Set<V> sinks, final int branchWeightThreshold) {
```

Comment:
Reminder to self: this `branchWeightThreshold` in the constructor is currently not actually hooked up, please fix when responding to the comments in the branch.